### ISSUE TITLE
On Off Switch: Not accessible

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#11225]
`pkg/lib/cockpit-components-onoff.jsx` is not accessible for the following reasons:

1. Clickable region is too small; it is limited to the toggle button, not the complete widget.
2. Lack of keyboard navigation & activation.
3. Lack of focus indicator (possibly related to lack of keyboard navigation).
4. Probably lacking various ARIA attributes for screen reader support.

**Comment by garrett** [comment]
PF3 Switch is based on "Bootstrap Switch" (which appears to be unmaintained?):
https://www.patternfly.org/pattern-library/widgets/#bootstrap-switch

PF4 Switch addresses the above concerns... except for a focus ring (which can be fixed via CSS). We could rewrite our React widget to be similar to PF4's:
https://pf4.patternfly.org/components/Switch/examples/

_**Note**: Some parts of Cockpit are not implemented in React, so we'd need to do more than just change the React widget. A quick search through our code indicates it's manually implemented. in `pkg/shell/index.html`, `pkg/networkmanager/index.html`, `pkg/systemd/index.html`, `pkg/playground/jquery-patterns.html`._

**Comment by martinpitt** [comment]
In PR #10451 I tried to move from our own On/Off swich to PatternFly-React's, but it's too buggy.

**Comment by garrett** [comment]
@martinpitt: It sounds like we'll have to roll our own version 2 of the widget.

I guess I'll wind up doing this, probably with your support. My plan would be to take the PF4 HTML as a starting point and make it look similar to what we have now.

FWIW, I also did a demo for PatternFly for a pure CSS label switch. Some ideas there could possibly be integrated too. Mostly the `: checked` and `~` sibling element so we can swap things around in CSS based on the widget state. This should make it simpler and easier to port around too.

**Comment by garrett** [comment]
Related: Animation for the switch @ https://github.com/cockpit-project/cockpit/issues/11226

I'll probably, eventually, work on a PR to address both this issue and the other at some point in the near future.

**Comment by garrett** [comment]
Here's a CodePen for my attempt to reimplement a switch similar to PF4, but using PF3 style (colors & fonts): https://codepen.io/garrett/pen/MRZYaV

It should be a lot more accessible already (it's a checkbox wrapped in a label), but we might need a touch of JS in our widget to toggle ARIA states (as PF4 does).

**Comment by martinpitt** [comment]
I turned that into actual code in PR #11769.
